src/mixins.coffee

```diff
@@ -64,3 +64,75 @@ Ember.Widgets.BodyEventListener = Ember.Mixin.create
   _removeDocumentHandlers: ->
     $(@get('bodyElementSelector')).off "click", @_clickHandler
     @_clickHandler = null
+
+Ember.Widgets.TabbableModal = Ember.Mixin.create
+  currentFocus: null
+
+  _focusTabbable: ->
+     # Set focus to the first match:
+     # 1. The saved focused element
+     # 2. First element inside the dialog matching [autofocus]
+     # 3. Tabbable element inside the content element
+     # 4. The close button (has class "close")
+    hasFocus = [@get 'currentFocus']
+    if hasFocus[0] == null or not $.contains(@$()[0], hasFocus[0])
+      hasFocus = @$( "[autofocus]" )
+    if hasFocus.length == 0
+      hasFocus = @$( ":tabbable" )
+    if hasFocus.length > 0
+      focusElement = hasFocus[0]
+      if focusElement.className.indexOf("close") > -1
+        # if we have more than two tabbable objects, we do not want to tab
+        # to close button
+        # while if we do not have any choice, the close button is chosen
+        if hasFocus.length > 1
+          focusElement = hasFocus[1]
+      focusElement.focus()
+      @set 'currentFocus', focusElement
+
+
+  _checkContainingElement: (containers, element) ->
+    for containerItem in containers
+      if containerItem is element or $.contains(containerItem, element)
+        return yes
+    return no
+
+  click: (event) ->
+    isActive = $.contains(@$()[0], event.target)
+    if not isActive
+      @hide() unless @get('enforceModality')
+      @_focusTabbable()
+    if not @_checkContainingElement(@$(':focusable'), event.target)
+      @_focusTabbable()
```

Comment:
It is a bit different: 
It will focus on the latest focused item (if there is one, otherwise, the first tabbable one, as you described) if we click on an element not tabbable.
But I get your point. I think it's a good idea. I will check with some people to see which one they like.